Further splitting down elementwise functions

oleksandr-pavlyk opened the pull request:
Added missing includes of `"common_inplace.hpp"` and split elementwise functions into smaller files. 

Will continue until building succeeds.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

oleksandr-pavlyk commented:
@ndgrigorian I think adding missing `#include "common_inplace.hpp"` did the trick, or maybe further splitting. I would continue splitting files to granularity of a single function, but then create `"elementwise_common.cpp"` file which exports initialization module that calls each initialization routine for each elementwise function.

ndgrigorian commented:
> @ndgrigorian I think adding missing `#include "common_inplace.hpp"` did the trick, or maybe further splitting. I would continue splitting files to granularity of a single function, but then create `"elementwise_common.cpp"` file which exports initialization module that calls each initialization routine for each elementwise function.

Great, I will go ahead and commit those changes to this branch, and after that we can merge this.

oleksandr-pavlyk commented:
@ndgrigorian I change `static inline` functions in `elementwise_functions.hpp`. I removed `static` qualifier and moved them into anonymous namespace per https://stackoverflow.com/questions/10876930/should-one-never-use-static-inline-function
